### PR TITLE
Added tinymce editor for inputs of type text

### PR DESCRIPTION
**Pull request by bojangles495**

### CODE REVIEW

File: lib/controllers/crud/find-documents-from-search-term.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+const debug = require('debug')('W2:WarpJS:crud:FindAssocTargetOptions');
```
Comment (danosaure) [inline lib/controllers/crud/find-documents-from-search-term.js]: Fix the string

File: public/libs/customLinksModal.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+CustomLinks_removeEntityFromSelectionList = function (relationship) {
+    console.log("Remove");
+    var id = $(this).attr('id').split(":")[1];
+    $(this).remove();
+    $active.entity.changed=true;
+}
+
+CustomLinks_addTargetToEntityModal = function (label, target) {
+    var elemID = "Content-DefaultTableView-Delete:"+target.id;
```
Comment (danosaure) [inline public/libs/customLinksModal.js]: This need to be eslint.

File: public/libs/customLinksModal.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+CustomLinks_removeEntityFromSelectionList = function (relationship) {
```
Comment (danosaure) [inline public/libs/customLinksModal.js]: change filename with dash.

Comment (danosaure) [inline public/libs/customLinksModal.js]: Change names to be valid (camelCase). Use `function <name>(relationship) {}` syntax.

File: public/libs/customLinksModal.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    console.log("Remove");
```
Comment (danosaure) [inline public/libs/customLinksModal.js]: remove (no pun intended)

File: lib/controllers/crud/find-documents-from-search-term.js (hunk 1 above)
Comment (danosaure) [inline lib/controllers/crud/find-documents-from-search-term.js]: This is probably more clear if it was: `search-for-target-entity` or something.

File: public/libs/customLinksModal.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    var id = $(this).attr('id').split(":")[1];
+    $(this).remove();
+    $active.entity.changed=true;
+}
+
+CustomLinks_addTargetToEntityModal = function (label, target) {
+    var elemID = "Content-DefaultTableView-Delete:"+target.id;
+    var elem = $("<li role='presentation' id='"+elemID+"'><a href='#'>"
```
Comment (danosaure) [inline public/libs/customLinksModal.js]: HTML attributes should be double quoted, so inverse single- and double-quotes.

File: templates/MonApp_DefaultViews_Page.hst [hunk 6]
```diff
@@ -24,7 +24,12 @@
                                     </label>
                                     <div class="col-sm-10">
                                         <!-- TBD: Use unique names for IDs -->
-                                        <input type="text" class="form-control" id="{{js:this.name;/}}Input">
+                                        {{if}}this.basicProperty[0].propertyType == "text"{{then}}
+
```
Comment (danosaure) [inline templates/MonApp_DefaultViews_Page.hst]: remove this blank line

File: templates/MonApp_WebAPI.hst [hunk 7]
```diff
@@ -51,6 +51,50 @@ $(document).ready(function () {
...
+            if (result.success) {
+                $active.relationship.queryResults = [];
+                $active.relationship.queryResultsCount = result.resultList[0].queryResultsCount;
+                result.resultList[0].queryResult.forEach (function (queryResult) {
+                    var source = get{{js:this.getDomain().name;/}}Admin().findOrAddToEntityCache(queryResult["_id"], queryResult.type, queryResult);
+                    $active.relationship.queryResults.push(source);
+                });
+                callback();
+                return result;
+            }
```
Comment (danosaure) [inline templates/MonApp_WebAPI.hst]: `} else {` or even better, remove the `else` since there is a `return` in the `if`.

Comment (bojangles495) [inline templates/MonApp_WebAPI.hst]: done

Comment (danosaure) [inline templates/MonApp_WebAPI.hst]: The else block/content is useful. It's just the `else` keyword that is not useful:

```javascript
if (something) {
  doSomething();
  return;
}
doSomethingElse();
```

File: lib/controllers/crud/search-for-target-entity.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+                if (arr && !mongoErr1 && !mongoErr2) {
+                    resultList.push({
+                        queryType: currentCommand.command,
+                        targetType: currentCommand.targetType,
+                        queryResult: arr,
+                        queryResultsCount: count,
+                        error: err,
+                        status: msg
+                    });
+                    msg = "Found " + count + " instances for Type=" + targetType + " in Domain " + domain;
```
Comment (danosaure) [inline lib/controllers/crud/search-for-target-entity.js]: msg = `Found ${count} instances for Type=${targetType} in Domain ${domain}`;

**Pull request merged**